Makefile

```diff
@@ -0,0 +1,30 @@
+.PHONY: all clean format build test dev install-pre-commit public-header
```

Comment:
Since we already have all targets in Earthly, I think it is better to add it to Earthfile instead of Makefile.
Earthly supports running local command using keyword `LOCALLY`

Example usage
```bash
❯ cat Earthfile
VERSION 0.7
 
clean:
    LOCALLY
    RUN echo "Deleting target and build directory"
    RUN rm -rf target build


❯ ls
build  target  1  2  3  Earthfile


❯ earthly +clean
 Init 🚀
...
 Build 🔧
—————————————————————————
              +clean | --> RUN echo "Deleting target and build directory"
              +clean | Deleting target and build directory
              +clean | --> RUN rm -rf target build
              output | --> exporting outputs
...
========================== 🌍 Earthly Build  ✅ SUCCESS ==========================

View logs at https://cloud.earthly.dev/builds/11309e1a-50d3-4804-b990-29301130c192


❯ ls
1  2  3  Earthfile
```

Reply:
Good point, thanks @expressvpn-mariappan-r ! I didn't know that Earthly can run local commands.